hudi-common/src/test/java/org/apache/hudi/common/testutils/HoodieTestDataGenerator.java

```diff
@@ -952,10 +952,15 @@ public List<HoodieRecord> generateUpdates(String instantTime, Integer n) throws
    */
   public List<HoodieRecord> generateUpdates(String instantTime, Integer n, String schemaStr) throws IOException {
     List<HoodieRecord> updates = new ArrayList<>();
+    Map<Integer, KeyPartition> existingKeys = existingKeysBySchema.get(schemaStr);
```

Comment:
we do have another method called generateUniqueUpdates. may be we can dedup and retain just 1 method.

Reply:
There are some cases where there are multiple updates for the same key expected, like when testing deduping within the ingestion. I will revert the changes here and just use the other method for these FGReader tests